src/features/references/editor/ReferenceDetailsCard.tsx

```diff
@@ -0,0 +1,261 @@
+import { createContext, Dispatch, ReactElement, SetStateAction, useCallback, useContext, useState } from 'react';
+
+import { CloseIcon, EditIcon } from '../../../components/icons';
+import { cx } from '../../../lib/cx';
+import { Author, ReferenceItem } from '../../../types/ReferenceItem';
+
+interface IEditingContext {
+  editing: boolean;
+  setEditing: Dispatch<SetStateAction<boolean>>;
+}
+
+const EditingContext = createContext<IEditingContext>({
+  editing: false,
+  setEditing: () => {
+    console.log('seting edtiting');
+  },
+});
+
+const returnFormatedAuthorsString = (authorsArray: Author[]): string => {
+  let authorsString = '';
+  authorsArray.forEach((a, index) => {
+    authorsString += (index > 0 ? ', ' : '') + a.fullName;
+  });
+  return authorsString;
+};
+
+const getContentString = (content: string | Author[] | undefined, id: string): string => {
+  if (typeof content == 'undefined') {
+    return '';
+  } else if (typeof content != 'string' && id === 'authors') {
+    return returnFormatedAuthorsString(content);
+  } else if (id === 'citationKey') {
+    return '[' + content.toString() + ']';
+  } else {
+    return content.toString();
+  }
+};
+
+const DataTextInput = ({
+  content,
+  id,
+  onChangeHandler,
+}: {
+  content: string;
+  id: string;
+  onChangeHandler: CallableFunction;
+}) => {
+  const [value, setValue] = useState(content);
+
+  return (
+    <input
+      className="w-full border bg-slate-50 px-2 py-0.5"
+      name={id}
+      type="text"
+      value={value}
+      onBlur={(evt) => {
+        onChangeHandler(id, evt.target.value);
+      }}
+      onChange={(evt) => {
+        setValue(evt.target.value);
+      }}
+    />
+  );
+};
+
+const TableDataCell = ({
+  contentData,
+  id,
+  onChangeHandler,
+  editable,
+}: {
+  contentData: string | Author[];
+  id: string;
+  onChangeHandler: CallableFunction;
+  editable: boolean;
+}) => {
+  const { editing } = useContext(EditingContext);
+
+  if (!editing || !editable || typeof contentData != 'string') {
+    return <UnEditableTableDataCell contentData={contentData} id={id} />;
+  } else {
+    return <EditableTableDataCell contentData={contentData} id={id} onChangeHandler={onChangeHandler} />;
+  }
+};
+
+const EditableTableDataCell = ({
+  contentData,
+  id,
+  onChangeHandler,
+}: {
+  contentData: string;
+  id: string;
+  onChangeHandler: CallableFunction;
+}) => (
+  <td className="w-auto p-5">
+    <DataTextInput content={contentData} id={id} onChangeHandler={onChangeHandler} />
+  </td>
+);
+
+const UnEditableTableDataCell = ({ contentData, id }: { contentData: string | Author[]; id: string }) => {
+  const contentString = getContentString(contentData, id);
+
+  return (
+    <td className="w-auto p-5">
+      <span className="leading-[30px]">{contentString}</span>
+    </td>
+  );
```

Comment:
`EditableTableDataCell` and `UneditableTableDataCell` are very similar and only used in `TableDataCell`. I think you should not separate the code in two components and write everything in `TableDataCell` directly instead, like this:
```typescript
return <td className="w-auto p-5">
    {editing && editable && typeof contentData === 'string' ?
      <DataTextInput content={contentData} id={id} onSave={onChangeHandler} /> :
      <span className="leading-[30px]">{getContentString(contentData, id)}</span>
    }
  </td>;
```